### PR TITLE
WIP: Updates for event-store v7

### PR DESCRIPTION
**Pull request by mablae**
What is done: 

- [x] Update packages to v7 dev versions
- [x] Make test pass with old behaivor
- [x] Updated CS-Fixer Config 
- [x] Fixed Travis build
- [x] Removed references to missing Classes (JsonPayloadSerializer)
- [x] Fix namespaces of moved stuff (EventSourcing)

### CODE REVIEW

**Comment by prolic** [comment]
Composer deps not suitable. See travis.

**Comment by mablae** [comment]
It's WIP - Only created the PR to show there is progress.

File: test/DependencyInjection/Fixture/config/yml/event_store.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 prooph_event_store:
   stores:
     main_store:
-      adapter: 'ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Adapter\BlackHole'
+      type: 'ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Adapter\BlackHole'
       repositories:
```
Comment (mablae) [inline test/DependencyInjection/Fixture/config/yml/event_store.yml]: True

File: composer.json [hunk 2]
```diff
@@ -37,23 +36,6 @@
     "DDD",
     "PHP"
   ],
-  "authors": [
```
Comment (prolic) [inline composer.json]: why no authors any more? please leave that and also add the new symfony maintainers

File: composer.json [hunk 3]
```diff
@@ -70,18 +52,21 @@
     }
   },
   "require": {
-    "php": "^7.0",
+    "php": "^7.1",
     "symfony/config": "~2.8 || ~3.0",
     "symfony/dependency-injection": "~2.8 || ~3.0",
     "symfony/http-kernel": "~2.8 || ~3.0",
-    "prooph/event-store": "^6.3"
+    "prooph/event-store": "^7.0"
```
Comment (prolic) [inline composer.json]: better use dev-7.x for now, the beta is not good.

File: composer.json [hunk 4]
```diff
@@ -70,18 +52,21 @@
...
   },
   "require-dev": {
+    "prooph/event-sourcing": "dev-develop as 5.0.0",
+    "prooph/snapshot-store": "dev-master",
+    "prooph/pdo-event-store": "dev-master",
     "phpunit/phpunit": "^5.3",
     "phpunit/php-invoker": "^1.1.4",
     "fabpot/php-cs-fixer": "^1.11",
     "bookdown/bookdown": "1.x-dev as 1.0.0",
```
Comment (prolic) [inline composer.json]: this should not be needed

File: composer.json [hunk 5]
```diff
@@ -70,18 +52,21 @@
...
     "symfony/dependency-injection": "~2.8 || ~3.0",
     "symfony/http-kernel": "~2.8 || ~3.0",
-    "prooph/event-store": "^6.3"
+    "prooph/event-store": "^7.0"
   },
   "require-dev": {
+    "prooph/event-sourcing": "dev-develop as 5.0.0",
+    "prooph/snapshot-store": "dev-master",
+    "prooph/pdo-event-store": "dev-master",
     "phpunit/phpunit": "^5.3",
```
Comment (prolic) [inline composer.json]: update to v6

File: composer.json [hunk 6]
```diff
@@ -70,18 +52,21 @@
...
     "phpunit/php-invoker": "^1.1.4",
```
Comment (prolic) [inline composer.json]: is this really needed explicitly?

File: composer.json [hunk 7]
```diff
@@ -70,18 +52,21 @@
...
     "fabpot/php-cs-fixer": "^1.11",
```
Comment (prolic) [inline composer.json]: we have prooph php-cs-fixer-config

File: src/EventStoreFactory.php [hunk 8]
```diff
@@ -21,25 +21,25 @@ class EventStoreFactory
...
     public function create(
         string $eventStoreName,
-        Adapter $adapter,
+        EventStore $type,
         ActionEventEmitter $actionEventEmitter,
         ContainerInterface $container,
         array $pluginServiceIds
     ): EventStore {
-        $eventStore = new EventStore($adapter, $actionEventEmitter);
+        $eventStore = new ActionEventEmitterEventStore($type, $actionEventEmitter);
```
Comment (prolic) [inline src/EventStoreFactory.php]: so ActionEventEmitterEventStore is always required? The interop config factories provided in event-store make this optional.

File: test/DependencyInjection/Fixture/Metadata/BlackHole.php [hunk 9]
```diff
@@ -11,6 +11,8 @@
 
 namespace ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Metadata;
 
+use Prooph\Common\Messaging\Message;
+
 class BlackHole implements \Prooph\EventStore\Metadata\MetadataEnricher
```
Comment (prolic) [inline test/DependencyInjection/Fixture/Metadata/BlackHole.php]: I think NoOpMetadataEnricher is a better name. Actually it does something.

File: src/EventStoreFactory.php (hunk 8 above)
Comment (mablae) [inline src/EventStoreFactory.php]: At least that's how I understood @codeliner . I may be wrong.

File: test/DependencyInjection/Fixture/Metadata/BlackHole.php (hunk 9 above)
Comment (mablae) [inline test/DependencyInjection/Fixture/Metadata/BlackHole.php]: Wasn't changed - not part of the PR ;-) But I get your point.

File: composer.json (hunk 3 above)
Comment (mablae) [inline composer.json]: 👍

File: composer.json (hunk 2 above)
Comment (mablae) [inline composer.json]: My fault. I think rebase ate it.

File: composer.json (hunk 6 above)
Comment (mablae) [inline composer.json]: I needed the eventsourcing interfaces for the mock objects. I dont like it either. 

After this PR there will follow more. It is not meant as a new stable yet. 
See discussion on prooph private chat today with @rabbl

File: test/DependencyInjection/Fixture/Plugin/GlobalBlackHole.php [hunk 10]
```diff
@@ -10,15 +10,18 @@
 
 namespace ProophTest\Bundle\EventStore\DependencyInjection\Fixture\Plugin;
 
-use Prooph\EventStore\EventStore;
+use Prooph\EventStore\ActionEventEmitterEventStore;
 use Prooph\EventStore\Plugin\Plugin;
 
 class GlobalBlackHole implements Plugin
 {
-    public $valid = false;
```
Comment (rabbl) [inline test/DependencyInjection/Fixture/Plugin/GlobalBlackHole.php]: I think this line could stay.

File: composer.json (hunk 5 above)
Comment (rabbl) [inline composer.json]: Should these three lines go to "require"?

File: src/EventStoreFactory.php (hunk 8 above)
Comment (mablae) [inline src/EventStoreFactory.php]: @codeliner @prolic 

@rabble and me will solve this in next PR.

File: src/DependencyInjection/Configuration.php [hunk 11]
```diff
@@ -26,13 +26,13 @@
      */
     public function __construct($debug)
     {
-        $this->debug = (Boolean)$debug;
+        $this->debug = (bool) $debug;
```
Comment (UFOMelkor) [inline src/DependencyInjection/Configuration.php]: Not part of this PR, but IMHO it would be cleaner to type hint against boolean and move the type casting to [`ProophEventStoreExtension:34`](https://github.com/mablae/event-store-symfony-bundle/blob/a093cef432dd924cae2c52f136536fb4f6068458/src/DependencyInjection/ProophEventStoreExtension.php#L34).

Comment (mablae) [inline src/DependencyInjection/Configuration.php]: I just removed it. It is not used at all.

**Comment by lunetics** [comment]
Is there any update on this pr?

**Comment by mablae** [comment]
@lunetics Sure. This will be worked on again, when Event-Store v7 has a stable release. 

Any feedback is very welcome! Thanks in advance.

**Comment by lunetics** [comment]
@mablae pretty new to prooph, trying that pr now in an project, so a fresh view on it

**Comment by prolic** [comment]
:+1:
